### ISSUE TITLE
Describing what the algorithms are helpful for with a brief visual and textual examples - a GSoC idea

### ISSUE THREAD

**Issue by stanyas** [networkx/networkx#5510]
### Current Behavior

The descriptions of algorithms [are missing](https://networkx.org/documentation/stable/reference/algorithms/), so it is hard to know which are suitable. If one approaches the NetworkX without thorough background knowledge of graphs, they may not understand which of them would be helpful without a lengthy comprehensive googling.

### Expected Behavior

There can be an overview of the high-level working of the graph algorithms with an image.

### Steps to Reproduce

For example, if one does not know what a _cut_ is, they will need to google when looking at [this example](https://networkx.org/documentation/stable/reference/algorithms/cuts.html). 

### Environment

Python version: any
NetworkX version: any

@MridulS , @dschult, @rossbar

**Comment by MridulS** [comment]
I do like this! Something like https://scikit-learn.org/stable/_static/ml_map.png but for network science algorithms could help direct folks towards the "right" kind of algorithm they are looking to work on :)

**Comment by rossbar** [comment]
> I do like this! Something like https://scikit-learn.org/stable/_static/ml_map.png but for network science algorithms could help direct folks towards the "right" kind of algorithm they are looking to work on :)

Agreed - the sklearn knowledge map is awesome and a great way to get a sense of available functionality in the library. Something similar for NetworkX would be great. It'd be *extra* awesome if we generated the knowledge map using... networkx :)

**Comment by stanyas** [comment]
Thanks, @MridulS @rossbar. 

I love the idea of metanetworkx map (MM) in style of scikit. I could possibly make the MM made in Networkx and explored in [Gephi](https://gephi.org/) also clickable and filterable like with [sigma.js](https://www.sigmajs.org/). The MM could also enable including tutorials or examples from the web in the "leaves" of the graph.  

What are the next steps with the application? Should I link here my proposal [like this](https://github.com/networkx/networkx/discussions/5111) for you to give feedback? Should I make an example pull request for a mini MM?  Can this be a 350 hours project?

Thanks.

**Comment by MridulS** [comment]
> What are the next steps with the application? Should I link here my proposal https://github.com/networkx/networkx/discussions/5111 for you to give feedback?

Yes that helps us suggest more things.

> Should I make an example pull request for a mini MM? 

That would be great!

> Can this be a 350 hours project given that I would need to understand, visualize, and provide a minimum viable code for each algorithm?

As these things are flexible you can choose to either submit a 175 hour or 350 hour proposal :)

**Comment by rossbar** [comment]
> I could possibly make the MM made in Networkx and explored in [Gephi](https://gephi.org/) also clickable and filterable like with [sigma.js](https://www.sigmajs.org/).

Just a quick comment here - I think it'd be a priority to keep as much of the analysis and visualization in Python/NetworkX as possible (this is a networkx project after all!) I think there are plenty of options for visualization that are better tied to Python, and specifically the scientific Python ecosystem. Just my two cents before you start committing too heavily to any particular approach!

**Comment by stanyas** [comment]
A tentative abstract:

NetworkX contains many algorithms. When I tried to have some specific functionality of a graph algorithm, I needed to do a lengthy googling of its behavior that would lead me to its name in NetworkX. 

The purpose of my project would thus be to demonstrate the functioning of each of the 60+ algorithms visually side by side in an image posted in a hyperlinked GitHub repository file. I would use the same graph to make the comparison easier. 

If some visualization would seem conducive to what the user wanted to do, they can search the file for its name or a number and see the next visualization layer of functions within the topic. If they chose an algorithm belonging to a class, they would get the code that generated the image and links to other projects that used the function.
---
I graphed the NetworkX of NetworkX, but it seems that the overlap between the classes is minimal, so that might not be an appropriate example on which to demonstrate the functioning. I also changed the image to suggest the working of the algorithm, but it did not work with the matplotlib zooming, and with 300+ nodes and edges, it would be unfeasible.

Tomorrow, I plan to make the pull request for this issue of two images of a simple graph to demonstrate what I meant in the abstract.
